### PR TITLE
Allow specifying DB engine in database config

### PR DESCRIPTION
**Pull request by djmetzle**
We're running a mysql engine RDS instance in production for strapi.

CC @dhmacs @lithobraking @danielbeardsley

### CODE REVIEW

**Review by danielbeardsley — APPROVED** [review]
Looks right, but I can't say for certain there isn't more needed.

File: backend/config/database.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ module.exports = ({ env }) => {
   if (env("NODE_ENV") === "production") {
     return {
       connection: {
-        client: "postgres",
+        client: "mysql",
```
Comment (djmetzle) [inline backend/config/database.js:7]: For now, lets just allow overriding Postgres from the ENV?
```suggestion
        client: env("DATABASE_ENGINE", "postgres"),
```

**Comment by djmetzle** [comment]
Kewl, so that passed all the tests. :+1:

**Comment by danielbeardsley** [comment]
CR :+1:

**Comment by jarstelfox** [comment]
CR 🌵

**Comment by dhmacs** [comment]
We've encountered [a few issues](https://github.com/mysqljs/mysql/pull/1962) with the integration between Strapi and mysql, so we're going to stick with postgres for the time being

**Pull request closed without merging**